var/spack/repos/builtin/packages/edm4hep/package.py

```diff
@@ -61,12 +61,20 @@ class Edm4hep(CMakePackage):
         description="Use the specified C++ standard when building.",
     )
 
+    variant(
+        "json",
+        default=True,
+        description="Build edm4hep with JSON support and edm4hep2json",
+        when="@0.99.2:",
+    )
+
     depends_on("cmake@3.3:", type="build")
     depends_on("cmake@3.23:", type="build", when="@0.10.3:")
     depends_on("python", type="build")
 
     depends_on("root@6.08:")
-    depends_on("nlohmann-json@3.10.5:")
+    depends_on("nlohmann-json@3.10.5:", when="@0.99.2: +json")
```

Comment:
1. Fix typo: The version range still starts with `@3.10.5:` (not only `3.10.5`):
2. As `nlohmann-json`, a header-only library, you could also use `type="build"` for it.

```suggestion
    with default_args(type="build"):
        depends_on("nlohmann-json@3.10.5:", when="@0.99.2: +json")
        depends_on("nlohmann-json@3.10.5:", when="@:0.99.1")
```